src/program.js

```diff
@@ -66,13 +67,21 @@ export class Program {
     argv = argv || process.argv.slice(2);
 
     // NOTE: always initialize yargs explicitly with the package dir
-    // so that we are sure that it is going to load the 'boolean-negation: false'
-    // config (See web-ext#469 for rationale).
+    // so that we are sure that it is going to load web-ext package.json
```

Comment:
The function signature suggests that `absolutePackageDir` is the current working dir, which makes the comment as a whole look a bit odd. I think that the comment would be a bit more clear if you also mention that `bin/web-ext` fixes the `absolutePackageDir` value to the `web-ext` project root.

And in this PR you've also removed the `yargs` field. Is this to ensure that we aren't unexpectedly loading additional rules? If so, modify the comment to emphasize that.

Reply:
> And in this PR you've also removed the `yargs` field.

For yargs field you mean the yargs property in the web-ext package.json file?

(I removed that because it is deprecated: https://github.com/yargs/yargs/blob/14920d13a0807e77fbbce9aa2370ec44b8a0e8d1/yargs.js#L1026)

Reply:
The comment refers to "web-ext package.json". What is its significance?

Reply:
That means that we want to be sure that yargs looks for the web-ext package.json file, instead of trying to discover a package.json file on its own by looking elsewhere (e.g. by looking for it in the current work dir or the package.json related to the currently running nodejs script).

While working on this I noticed that yargs is actually not always using the directory we pass it (e.g. it is not using that parameter [here](https://github.com/yargs/yargs/blob/14920d13a0807e77fbbce9aa2370ec44b8a0e8d1/yargs.js#L824) and [here](https://github.com/yargs/yargs/blob/14920d13a0807e77fbbce9aa2370ec44b8a0e8d1/yargs.js#L1024)), and so in some cases it is still looking for a package.json that could not be the one that we want to point yargs to (and this is making yargs to emit the warning log I linked above while running the unit tests, because yargs is looking to the "yargs" section in the mocha package.json).

The above issue doesn't seem to be triggering any problems right now (because during the regular web-ext usage and functional tests, the current nodejs script is `bin/web-ext`, and so yargs is looking for the right package.json file in that case, and the issue is not currently triggering any unexpected behavior in the unit tests), but I'm going to open an issue / pull request on yargs to discuss this unexpected yargs behavior.

Reply:
As currently written, it seems that the web-ext package.json has some special config. With your explanation, it's clearer, but please update the comment to emphasize that our `package.json` has no config, and that this logic here is meant to avoid having side effects that would occur when loaded from an arbitrary directory, such as when run from the `mocha` directory, referencing 469 for rationale.